catalyst-gateway/bin/src/settings/cassandra_db.rs

```diff
@@ -0,0 +1,121 @@
+//! Command line and environment variable settings for the service
+
+use tracing::info;
+
+use super::str_env_var::StringEnvVar;
+use crate::db::{
+    self,
+    index::session::{CompressionChoice, TlsChoice},
+};
+
+/// Default Cassandra DB URL for the Persistent DB.
+pub(super) const PERSISTENT_URL_DEFAULT: &str = "127.0.0.1:9042";
+
+/// Default Cassandra DB URL for the Persistent DB.
+pub(super) const PERSISTENT_NAMESPACE_DEFAULT: &str = "persistent";
+
+/// Default Cassandra DB URL for the Persistent DB.
+pub(super) const VOLATILE_URL_DEFAULT: &str = "127.0.0.1:9042";
+
+/// Default Cassandra DB URL for the Persistent DB.
+pub(super) const VOLATILE_NAMESPACE_DEFAULT: &str = "volatile";
+
+/// Default maximum batch size.
+/// This comes from:
+/// <https://docs.aws.amazon.com/keyspaces/latest/devguide/functional-differences.html#functional-differences.batch>
+/// Scylla may support larger batches for better performance.
+/// Larger batches will incur more memory overhead to store the prepared batches.
+const MAX_BATCH_SIZE_DEFAULT: i64 = 30;
+
+/// Minimum possible batch size.
+pub(crate) const MIN_BATCH_SIZE: i64 = 1;
+
+/// Maximum possible batch size.
+const MAX_BATCH_SIZE: i64 = 256;
+
+/// Configuration for an individual cassandra cluster.
+#[derive(Clone)]
+pub(crate) struct EnvVars {
```

Comment:
I would rather rename this struct to something like `CassandraClusterSettings` or `CassandraClusterConfig` or even maybe just `Config`.
`EnvVars` does not really represent what is it, and from the such name it is not clear what is the intention of this object.

Reply:
It is 100% env vars, they are config sure, but they only come from EnvVars, nowhere else?
Naming it CassandraClusterSettings is redundant when it's inside a cassandra_db file and namespaced as such.
It could be called `Config` but I don't think thats any better or worse than EnvVars, especially when the config in this case can only be controlled by EnvVars.

This type is always referred to as `cassandra_db::EnvVars`.    It used to be like you suggested and it was ugly and didn’t help comprehension at all.  It just meant there were a lot more `uses` items in a file to manage.